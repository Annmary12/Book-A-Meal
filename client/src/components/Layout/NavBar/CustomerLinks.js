import React, { Fragment } from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import styles from './navbar.scss';

const CustomerLinks = ({ pathname }) => (
  <Fragment>
    <li>
      <Link
        className={classNames({
          [styles.active]: pathname === '/menus'
        })}
        to="/menus"
      >
        Menus
      </Link>
    </li>
    <li>
      <Link
        className={classNames({
          [styles.active]: pathname === '/orders'
        })}
        to="/orders"
      >
        Order history
      </Link>
    </li>
  </Fragment>
);

export default CustomerLinks;
