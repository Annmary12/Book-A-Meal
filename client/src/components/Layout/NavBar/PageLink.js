import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './navbar.scss';

const PageLink = ({ pathname, to, text }) => (
  <Link
    className={classNames({
      [styles.active]: pathname === to
    })}
    to={to}
  >
    {text}
  </Link>
);

PageLink.propTypes = {
  pathname: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default PageLink;
