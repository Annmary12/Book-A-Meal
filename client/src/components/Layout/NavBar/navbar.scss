.nav{
  background-color: #d24e24;
  padding: 9px;
  color: white;

  @media only screen and (max-width: 600px){
    position: absolute;
    padding: 9px;
    color: white;
    margin-bottom: 0px;
    width: 96%;
  }


  @media only screen and (min-width: 600px) {
    display: flex;
    flex-direction: row;
    padding-left: 30px;
    padding-right: 30px;

    /*background-color: #a82800;*/
    margin-bottom: 0px;
    height: 40px;

  }
}

.titleArea{
  margin-top: 4px;
  @media only screen and (max-width: 600px){
    display: flex;
		width: 100%;
  }
}

.title{
	font-family: "Dancing Script",cursive, calibri;
	font-size: 30px;
  font-weight: bold;
  position: absolute;
  left: 69px;
  text-decoration: none;
  color: white;
  text-align: center;

  @media only screen and (max-width: 600px){
    position: absolute;
    top:none;
    left: none;
    text-decoration: none;
    color: white;
  }
}

.logo{
  position: absolute;
  left: -50px;
}

.appName{
  @media only screen and (max-width: 600px){
    width: 190px;
  }
}

.navMenus{
  margin-left: auto;

  @media only screen and (max-width: 600px) {
    display: none;
  }
}

.navList{
  list-style: none;
  text-align: center;
  padding: 0;
  width: 100%;
  margin: auto;


  @media only screen and (max-width: 600px) {
    margin-top: 8px;
  }

  li{
    display: inline-block;
		margin-right: 4px;
    cursor: pointer;

    @media only screen and (max-width: 600px) {
      border-bottom: 1px solid #fbfbfb;
      display: block;
    }

    a{
      border-radius: 5px;
		  font-size: 18px;
      color: white;
      text-decoration: none;
      display: block;
      padding: 8px;

      &:hover{
        background-color: #902605;
        border: 3px dotted #71230b;
      }
    }
  }
}

.active{
  background-color: #902605;
  border: 3px dotted #71230b;

  @media only screen and (max-width: 600px) {
    margin-bottom: 4px;
    border: 2.5px dotted #71230b;
  }
}
.visible{
  display: block;
  @media only screen and (max-width: 600px) {
    :global{
      animation: fadeInDown .14s linear;
    }
  }
}

.hidden{
  display: none;
}

.home{
  background-color: #fd49119e;
  box-shadow: 0 120px 320px rgba(0, 0, 0, 0.41), 0 15px 12px rgba(0,0,0,0.22);
}

.navToggle{
  display: none;
  outline: none;

  @media only screen and (max-width: 600px){
    display: block;
    background-color: transparent;
    border-radius: 4px;
    border-style: solid;
    border-color: brown;
    color: wheat;
    font-size: 21px;
    width: 45px;
    margin-left: auto;
    height: 36px;
  }
}