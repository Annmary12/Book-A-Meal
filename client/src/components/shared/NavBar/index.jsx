import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import logo from '../../../../assets/img/logo-white.svg';
import styles from './style.scss';

/**
 * @class NavBar
 */
class NavBar extends Component {
  static propTypes = {
    backgroundColor: PropTypes.string
  }
  static defaultProps = {
    backgroundColor: '#e54310'
  }
  state = {
    navMenuStyle: {}
  }
  /**
   * @returns {undefined} - undefined
   */
  toggleMenu() {
    const visible = { display: 'block' };
    const hidden = { display: 'none' };
    this.setState({
      navMenuStyle: this.state.navMenuStyle.display === 'block'
        ? hidden
        : visible
    });
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <nav className={styles.nav} style={{ backgroundColor: this.props.backgroundColor }}>
        <div className={styles['title-area']}>
          <Link to="/" className={styles.title}>
            <img className={styles.logo} src={logo} width="30" alt="logo" />
            <span className={styles['app-name']}>Book-A-Meal</span>
          </Link>
          <button id="nav-toggle" onClick={() => this.toggleMenu()} href="#">&#9776;</button>
        </div>
        <div className={styles['nav-menus']} style={this.state.navMenuStyle}>
          <ul className={styles['nav-list']}>
            <li>
              <Link to="/login">Login</Link>
            </li>
            <li>
              <Link to="/signup">Sign Up</Link>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default NavBar;
