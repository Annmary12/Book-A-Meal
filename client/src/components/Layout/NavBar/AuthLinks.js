import React, { Fragment } from 'react';
import CustomerLinks from './CustomerLinks.js';
import { Link } from 'react-router-dom';

const AuthLinks = ({ role, pathname }) => {
  //if (role === 'customer') {
    return (
      <CustomerLinks
        pathname={pathname}
      />
    );
  //}
};

export default AuthLinks;
