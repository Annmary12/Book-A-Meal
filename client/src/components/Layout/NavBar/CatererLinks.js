import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import PageLink from './PageLink.js';

const CatererLinks = ({ pathname }) => (
  <Fragment>
    <li>
      <PageLink to="/manage-meals" text="Meals" pathname={pathname} />
    </li>
    <li>
      <PageLink to="/manage-menus" text="Menus" pathname={pathname} />
    </li>
    <li>
      <PageLink to="/manage-orders" text="Orders" pathname={pathname} />
    </li>
  </Fragment>
);

CatererLinks.propTypes = {
  pathname: PropTypes.string.isRequired
};

export default CatererLinks;
