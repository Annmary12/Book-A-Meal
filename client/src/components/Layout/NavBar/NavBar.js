import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import AuthLinks from './AuthLinks.js';
import logo from '../../../../assets/img/logo-white.svg';
import styles from './navbar.scss';

/**
 * @class NavBar
 */
class NavBar extends Component {
  static propTypes = {
    home: PropTypes.bool,
    loggedIn: PropTypes.bool,
    logout: PropTypes.func.isRequired,
    pathname: PropTypes.string.isRequired,
    user: PropTypes.objectOf(PropTypes.string).isRequired
  }
  static defaultProps = {
    home: false,
    loggedIn: false
  }
  state = {
    isNavMenuVisible: false
  }

  /**
   * @returns {undefined} - undefined
   */
  toggleMenu() {
    this.setState({
      isNavMenuVisible: !this.state.isNavMenuVisible
    });
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    const { pathname, user } = this.props;

    return (
      <nav className={classNames({
        [styles.nav]: true,
        [styles.home]: this.props.home
      })}
      >
        <div className={styles.titleArea}>
          <Link to="/" className={styles.title}>
            <img className={styles.logo} src={logo} width="30" alt="logo" />
            <span className={styles.appName}>Book-A-Meal</span>
          </Link>
          <button className={styles.navToggle} onClick={() => this.toggleMenu()} href="#">
            { this.state.isNavMenuVisible ?
              <span style={{ fontSize: 29 }}>&times;</span> :
              <span>&#9776;</span>
            }
          </button>
        </div>
        <div
          className={classNames({
            [styles.navMenus]: true,
            [styles.visible]: this.state.isNavMenuVisible
          })}
        >
          <ul className={styles.navList}>
            {
              this.props.loggedIn ?
                <Fragment>
                  <AuthLinks
                    role={user.role}
                    pathname={pathname}
                  />
                  <li>
                    <Link to="/" onClick={this.props.logout}>Logout</Link>
                  </li>
                </Fragment> :
                <Fragment>
                  <li>
                    <Link to="/login">Login</Link>
                  </li>
                  <li>
                    <Link to="/signup">Sign Up</Link>
                  </li>
                </Fragment>
            }
          </ul>
        </div>
      </nav>
    );
  }
}

export default NavBar;
