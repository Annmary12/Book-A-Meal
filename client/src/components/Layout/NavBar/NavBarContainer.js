import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { logout } from '../../../actions/auth.js';
import NavBar from './NavBar.js';

const mapStateToProps = state => ({
  loggedIn: state.auth.loggedIn,
  user: state.auth.user,
  pathname: state.router.location.pathname
});

const mapDispatchToProps = dispatch => ({
  logout: bindActionCreators(logout, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
