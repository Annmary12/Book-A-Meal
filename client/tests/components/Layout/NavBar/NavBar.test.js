import React from 'react';
import NavBar from '../../../../src/components/Layout/NavBar/NavBar';

describe('NavBar component', () => {
  const props = {
    home: true,
    loggedIn: true,
    logout: jest.fn(),
    pathname: '/test',
    user: {
      id: 'test-id',
      name: 'Otse Obande',
      email: 'otseobande@gmail.com',
      username: 'otseobande'
    }
  }
  it('should match snapshot if loggedIn props is true', () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavBar
          {...props}
          loggedIn={true}
        />
      </MemoryRouter>
    );
    const navBarWrapper = wrapper.find('NavBar');

    expect(navBarWrapper).toMatchSnapshot();
  });

  it('should match snapshot if loggedIn props is false', () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavBar
          {...props}
          loggedIn={false}
        />
      </MemoryRouter>
    );
    const navBarWrapper = wrapper.find('NavBar');

    expect(navBarWrapper).toMatchSnapshot();
  });

  it('should match snapshot if state.isNavMenuVisible is true', () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavBar
          {...props}
          loggedIn={false}
        />
      </MemoryRouter>
    );
    const navBarWrapper = wrapper.find('NavBar');

    navBarWrapper.instance().setState({isNavMenuVisible: true});

    expect(navBarWrapper).toMatchSnapshot();
  });

  describe('toggleMenu method', () => {
    it('should toggle isNavMenuVisible when called', () => {
      const wrapper = mount(
        <MemoryRouter>
          <NavBar
            {...props}
            loggedIn={false}
          />
        </MemoryRouter>
      );
      const navBarWrapper = wrapper.find('NavBar');

      navBarWrapper.instance().setState({ isNavMenuVisible: false });
      navBarWrapper.instance().toggleMenu();
      
      expect(navBarWrapper.instance().state.isNavMenuVisible).toBeTruthy();
      navBarWrapper.instance().toggleMenu();
      expect(navBarWrapper.instance().state.isNavMenuVisible).toBeFalsy();
    });
  });
});