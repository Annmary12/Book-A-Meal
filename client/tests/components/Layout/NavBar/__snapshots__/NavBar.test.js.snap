// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match snapshot if loggedIn props is false 1`] = `
<NavBar
  home={true}
  loggedIn={false}
  logout={[MockFunction]}
  pathname="/test"
  user={
    Object {
      "email": "otseobande@gmail.com",
      "id": "test-id",
      "name": "Otse Obande",
      "username": "otseobande",
    }
  }
>
  <nav
    className="nav home"
  >
    <div
      className="titleArea"
    >
      <Link
        className="title"
        replace={false}
        to="/"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="logo"
            src="test-file-stub"
            width="30"
          />
          <span
            className="appName"
          >
            Book-A-Meal
          </span>
        </a>
      </Link>
      <button
        className="navToggle"
        href="#"
        onClick={[Function]}
      >
        <span>
          ☰
        </span>
      </button>
    </div>
    <div
      className="navMenus"
    >
      <ul
        className="navList"
      >
        <li>
          <Link
            replace={false}
            to="/login"
          >
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/signup"
          >
            <a
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</NavBar>
`;

exports[`NavBar component should match snapshot if loggedIn props is true 1`] = `
<NavBar
  home={true}
  loggedIn={true}
  logout={[MockFunction]}
  pathname="/test"
  user={
    Object {
      "email": "otseobande@gmail.com",
      "id": "test-id",
      "name": "Otse Obande",
      "username": "otseobande",
    }
  }
>
  <nav
    className="nav home"
  >
    <div
      className="titleArea"
    >
      <Link
        className="title"
        replace={false}
        to="/"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="logo"
            src="test-file-stub"
            width="30"
          />
          <span
            className="appName"
          >
            Book-A-Meal
          </span>
        </a>
      </Link>
      <button
        className="navToggle"
        href="#"
        onClick={[Function]}
      >
        <span>
          ☰
        </span>
      </button>
    </div>
    <div
      className="navMenus"
    >
      <ul
        className="navList"
      >
        <AuthLinks
          pathname="/test"
        >
          <CatererLinks
            pathname="/test"
          >
            <li>
              <PageLink
                pathname="/test"
                text="Meals"
                to="/manage-meals"
              >
                <Link
                  className=""
                  replace={false}
                  to="/manage-meals"
                >
                  <a
                    className=""
                    href="/manage-meals"
                    onClick={[Function]}
                  >
                    Meals
                  </a>
                </Link>
              </PageLink>
            </li>
            <li>
              <PageLink
                pathname="/test"
                text="Menus"
                to="/manage-menus"
              >
                <Link
                  className=""
                  replace={false}
                  to="/manage-menus"
                >
                  <a
                    className=""
                    href="/manage-menus"
                    onClick={[Function]}
                  >
                    Menus
                  </a>
                </Link>
              </PageLink>
            </li>
            <li>
              <PageLink
                pathname="/test"
                text="Orders"
                to="/manage-orders"
              >
                <Link
                  className=""
                  replace={false}
                  to="/manage-orders"
                >
                  <a
                    className=""
                    href="/manage-orders"
                    onClick={[Function]}
                  >
                    Orders
                  </a>
                </Link>
              </PageLink>
            </li>
          </CatererLinks>
        </AuthLinks>
        <li>
          <Link
            onClick={[MockFunction]}
            replace={false}
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Logout
            </a>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</NavBar>
`;

exports[`NavBar component should match snapshot if state.isNavMenuVisible is true 1`] = `
<NavBar
  home={true}
  loggedIn={false}
  logout={[MockFunction]}
  pathname="/test"
  user={
    Object {
      "email": "otseobande@gmail.com",
      "id": "test-id",
      "name": "Otse Obande",
      "username": "otseobande",
    }
  }
>
  <nav
    className="nav home"
  >
    <div
      className="titleArea"
    >
      <Link
        className="title"
        replace={false}
        to="/"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="logo"
            src="test-file-stub"
            width="30"
          />
          <span
            className="appName"
          >
            Book-A-Meal
          </span>
        </a>
      </Link>
      <button
        className="navToggle"
        href="#"
        onClick={[Function]}
      >
        <span>
          ☰
        </span>
      </button>
    </div>
    <div
      className="navMenus"
    >
      <ul
        className="navList"
      >
        <li>
          <Link
            replace={false}
            to="/login"
          >
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/signup"
          >
            <a
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</NavBar>
`;
